--- a/Makefile.in	2011-07-16 12:05:27.728570456 +0400
+++ b/Makefile.in	2011-07-16 12:06:24.945570476 +0400
@@ -3,6 +3,7 @@
 ##   static
 LIBRARY_TYPE=relocatable
 DESTDIR ?= 
+GNATFLAGS ?=
 # Replace with the location for the installation directory
 INSTALL=@INSTALL@
 PACKAGE_TARNAME=@PACKAGE_TARNAME@
@@ -20,7 +21,7 @@
 
 # Set a default value (ignored if set in the environment)
 PROCESSORS?=1
-GNATMAKE=@gnatmake@ -j${PROCESSORS}
+GNATMAKE=@gnatmake@  ${GNATFLAGS}
 
 # Compiler mode: one of "distrib", "debug", "optimize", "profile"
 MODE=distrib
