--- a/distrib/xmlada_dom.gpr	2011-03-01 12:18:26.928536602 +0300
+++ b/distrib/xmlada_dom.gpr	2011-03-01 12:17:54.631541017 +0300
@@ -1,12 +1,13 @@
+with "directories";
 with "xmlada_shared";
 with "xmlada_sax";
 project XmlAda_Dom is
     for Library_Name use "xmlada_dom";
 
     for Library_Kind use Xmlada_Shared.Xmlada_Kind;
-    for Source_Dirs use ("../../include/xmlada");
+    for Source_Dirs use (Directories.Includedir & "/xmlada");
     for Source_List_File use "xmlada/" & Project'Library_Name & ".lgpr";
-    for Library_Dir use "../xmlada/" & Project'Library_Kind;
+    for Library_Dir use Directories.Libdir & "xmlada/" & Project'Library_Kind;
     for Library_Version use
        "lib" & Project'Library_Name & ".so." & Xmlada_Shared.Version;
     for Externally_Built use "true";
--- a/distrib/xmlada_sax.gpr	2011-03-01 12:18:26.928536602 +0300
+++ b/distrib/xmlada_sax.gpr	2011-03-01 12:17:54.633540891 +0300
@@ -1,3 +1,4 @@
+with "directories";
 with "xmlada_shared";
 with "xmlada_input";
 with "xmlada_unicode";
@@ -5,9 +6,9 @@
     for Library_Name use "xmlada_sax";
 
     for Library_Kind use Xmlada_Shared.Xmlada_Kind;
-    for Source_Dirs use ("../../include/xmlada");
+    for Source_Dirs use (Directories.Includedir & "/xmlada");
     for Source_List_File use "xmlada/" & Project'Library_Name & ".lgpr";
-    for Library_Dir use "../xmlada/" & Project'Library_Kind;
+    for Library_Dir use Directories.Libdir & "xmlada/" & Project'Library_Kind;
     for Library_Version use
        "lib" & Project'Library_Name & ".so." & Xmlada_Shared.Version;
     for Externally_Built use "true";
--- a/distrib/xmlada_input.gpr	2011-03-01 12:18:26.928536602 +0300
+++ b/distrib/xmlada_input.gpr	2011-03-01 12:17:54.632540953 +0300
@@ -1,12 +1,13 @@
+with "directories";
 with "xmlada_shared";
 with "xmlada_unicode";
 project XmlAda_Input is
     for Library_Name use "xmlada_input_sources";
 
     for Library_Kind use Xmlada_Shared.Xmlada_Kind;
-    for Source_Dirs use ("../../include/xmlada");
+    for Source_Dirs use (Directories.Includedir & "/xmlada");
     for Source_List_File use "xmlada/" & Project'Library_Name & ".lgpr";
-    for Library_Dir use "../xmlada/" & Project'Library_Kind;
+    for Library_Dir use Directories.Libdir & "xmlada/" & Project'Library_Kind;
     for Library_Version use
        "lib" & Project'Library_Name & ".so." & Xmlada_Shared.Version;
     for Externally_Built use "true";
--- a/distrib/xmlada.gpr	2011-03-01 12:18:26.928536602 +0300
+++ b/distrib/xmlada.gpr	2011-03-01 12:17:54.632540953 +0300
@@ -1,3 +1,4 @@
+with "directories";
 --  Project file for use of XML/Ada.
 --  Put the directory that contains xmlada.gpr in your ADA_PROJECT_PATH
 --  variable. Then, in our own project file, add a "with" dependency on
--- a/distrib/xmlada_schema.gpr	2011-03-01 12:18:26.928536602 +0300
+++ b/distrib/xmlada_schema.gpr	2011-03-01 12:17:54.633540891 +0300
@@ -1,12 +1,13 @@
+with "directories";
 with "xmlada_shared";
 with "xmlada_dom";
 project XmlAda_Schema is
     for Library_Name use "xmlada_schema";
 
     for Library_Kind use Xmlada_Shared.Xmlada_Kind;
-    for Source_Dirs use ("../../include/xmlada");
+    for Source_Dirs use (Directories.Includedir & "/xmlada");
     for Source_List_File use "xmlada/" & Project'Library_Name & ".lgpr";
-    for Library_Dir use "../xmlada/" & Project'Library_Kind;
+    for Library_Dir use Directories.Libdir & "xmlada/" & Project'Library_Kind;
     for Library_Version use
        "lib" & Project'Library_Name & ".so." & Xmlada_Shared.Version;
     for Externally_Built use "true";
--- a/distrib/xmlada_unicode.gpr	2011-03-01 12:18:26.928536602 +0300
+++ b/distrib/xmlada_unicode.gpr	2011-03-01 12:17:54.633540891 +0300
@@ -1,11 +1,12 @@
+with "directories";
 with "xmlada_shared";
 project XmlAda_Unicode is
     for Library_Name use "xmlada_unicode";
 
     for Library_Kind use Xmlada_Shared.Xmlada_Kind;
-    for Source_Dirs use ("../../include/xmlada");
+    for Source_Dirs use (Directories.Includedir & "/xmlada");
     for Source_List_File use "xmlada/" & Project'Library_Name & ".lgpr";
-    for Library_Dir use "../xmlada/" & Project'Library_Kind;
+    for Library_Dir use Directories.Libdir & "xmlada/" & Project'Library_Kind;
     for Library_Version use
        "lib" & Project'Library_Name & ".so." & Xmlada_Shared.Version;
     for Externally_Built use "true";
